template/logging.tftpl

```diff
@@ -3,48 +3,53 @@ echo 'installing additional software for logging'
 for i in {1..7}
 do
   echo "Attempt: ---- " $i
-  yum install -y aws-cli awslogs jq && break || sleep 60
+  yum install -y aws-cli rsyslog amazon-cloudwatch-agent jq && break || sleep 60
 done
 
-# Inject the CloudWatch Logs configuration file contents
-cat > /etc/awslogs/awslogs.conf <<- EOF
-[general]
-state_file = /var/lib/awslogs/agent-state
-
-[/var/log/dmesg]
-file = /var/log/dmesg
-log_stream_name = {instanceId}/dmesg
-log_group_name = ${log_group_name}
-initial_position = start_of_file
-
-[/var/log/messages]
-file = /var/log/messages
-log_stream_name = {instanceId}/messages
-log_group_name = ${log_group_name}
-datetime_format = %b %d %H:%M:%S
-initial_position = start_of_file
-
-[/var/log/user-data.log]
-file = /var/log/user-data.log
-log_stream_name = {instanceId}/user-data
-log_group_name = ${log_group_name}
-initial_position = start_of_file
+# Create the CloudWatch Logs configuration file
+cat > /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json  <<- EOF 
+{
+    "agent": {
+        "run_as_user": "root"
+    },
+    "metrics": {
+        "metrics_collected": {
+            "cpu": {
+                "measurement": [
+                    "cpu_usage_idle"
+                ],
+                "metrics_collection_interval": 300,
+                "totalcpu": true
+            },
+            "mem": {
+                "measurement": [
+                    "mem_used_percent"
+                ],
+                "metrics_collection_interval": 300
+            }
+        }
+    },
+    "logs": {
+        "logs_collected": {
+            "files": {
+                "collect_list": [
+                    {
+                        "file_path": "/var/log/messages",
+                        "log_group_name": "${log_group_name}",
+                        "log_stream_name": "{instance_id}/messages"
+                    },
+                    {
+                        "file_path": "/var/log/user-data.log",
+                        "log_group_name": "${log_group_name}",
+                        "log_stream_name": "{instance_id}/user-data"
+                    }
+                ]
+            }
+        }
+    }
+}
```

Comment:
praise: Nice! Having a Json configuration now makes sense to me and improves the readability.